gems/smithy/lib/smithy/anvil/views/client/module.rb

```diff
@@ -38,7 +38,7 @@ def requires
             if @plan.type == :types
               [:types]
             else
-              [:client, :errors, :types]
+              ['plugins/endpoint', :client, :errors, :types, :endpoint_parameters, :endpoint_provider]
```

Comment:
May as well make this all strings. Can you alphabetize?

Reply:
Changed to all strings, but the order matters here so I can't alphabetize.  specially, the plugin must come before client.  I chose to do it this way to keep it simple.  The alternative is to come up with a codegen time dependency modeling.